### ISSUE TITLE
gutter symbols not displaying on second vertical split pane

### ISSUE THREAD

**Issue by dtrietsch** [mhinz/vim-signify#208]
I noticed today that gutter symbols are not displaying on second vertical split pane. vim-gitgutter is working when I swap it in.

**Comment by mhinz** [comment]
Can you give a few more details? You have 2 windows, both showing version-controlled files but one window doesn't sport any signs (although there are changes)?

**Comment by dtrietsch** [comment]
Correct, 2 windows, both showing version-controlled files, one doesn't have any signs (although there are changes). Yes, Same VCS (git).

Yes, SignifyList does list both files; however, the stats are wrong. (0, 0, 0 - I have some added, some changed, and some removed). SignifyDebug command is for SVN even though I'm git? Error is standard SVN file not a working copy, but, as I said, it's a git repo.

**Comment by mhinz** [comment]
`:SignifyDebug` should try all supported VCS, limited by `:h g:signify_vcs_list`.

Can you check that again? The git section would be enough. :)

**Comment by dtrietsch** [comment]
Output of SignifyDebug (as much as I can share):
git diff --no-color --no-ext-diff -U0 -- 'myFile.h'
================================================================  
<the output that follows looks good to me>
<then immediately followed by:>
svn diff --diff-cmd 'diff' -x -U0 -- '<fullPath>/myFile.h'
====================================================================================================================== 
svn: E155007: '<fullPath>/myFile.h' is not a working copy 
Press ENTER or type a command to continue

**Comment by mhinz** [comment]
What about the other 2 points?

**Comment by dtrietsch** [comment]
If "other 2 points" is the file that is in the left pane displaying properly, then the output of SignifyDebug is the same.

**Comment by mhinz** [comment]
https://github.com/mhinz/vim-signify/issues/208#issuecomment-274665979 should list 3 points (unless some browser chaching is going on).

**Comment by dtrietsch** [comment]
Oops... yeah, sorry, browser caching issue.

**Comment by dtrietsch** [comment]
1. SignifyList shows both buffers as git.
2. (see prior response)
3. Hmmm... running :verb w fixes the missing signs.
Says:
"myFile.h" 25L, 564C written
Writing undo file: name of file
[sy] Updating signs.
[sy:git] get_diff_start()

**Comment by dtrietsch** [comment]
Maybe I need to restart my machine... it's been a long time. I'll let you know if that works.

**Comment by mhinz** [comment]
Hmm, okay. So all points look okay and `:w` actually does what we want.

So it just does _not_ happen right after opening the buffer?

When happens if you do this:

```
$ vim
:e file1
:vs file2
```

Does it show signs for both windows?

**Comment by dtrietsch** [comment]
> $ vim
> :e file1
> :vs file2

Did not work.

No luck on the restart either.

**Comment by dtrietsch** [comment]
I'm using today's macvim (maybe that's the problem). I use Spf13, so maybe there's another Bundle causing a problem.

**Comment by mhinz** [comment]
When you say "did not work" you mean only one (the first) window showed buffers?

I'll install MacVim in the meanwhile.

**Comment by dtrietsch** [comment]
> you mean only one (the first) window showed buffers?

Actually, there may be another issue on top of what I'm describing if, upon opening buffer, I should see the signs. Here are the facts: 1) when I open the left buffer and the right buffer both do not show signs. 2) When I modify the left buffer, I see signs. 3) When I modify the right buffer, I don't see signs. 4) When I run :verb w twice on either buffer, I see signs. 5) When I run :verb w once, I don't see signs. 6) When I open the buffers with :e and :vs, I don't see signs.

**Comment by dtrietsch** [comment]
I gotta drive home, so I won't be able to respond for a little bit.

**Comment by mhinz** [comment]
Hmmm. I can reproduce some of the things you mentioned in MacVim. I'll play around with it a bit. Thanks for the responses so far!

**Comment by mhinz** [comment]
So, apparently the original MacVim was forked some time ago: https://github.com/macvim-dev/macvim/issues

Unfortunately it seems to have some weird rendering issues. If you use the same binary without GUI, it works without problems.

You can reproduce it when run `/Applications/MacVim.app/Contents/MacOS/Vim` with `Terminal.app` or `iTerm2`.

**Comment by mhinz** [comment]
I played around with some code that uses the `close_cb` handler (as gitgutter does) instead of the `exit_cb` handler as we do. This makes the signs work when you open a buffer, but also leads to more fragile code.

But then, just refreshing signs doesn't work (same issue with gitgutter). After `:w`, I first have to move the cursor so that the screen gets refreshed.

Putting it in a nutshell, MacVim is currently broken and there's no good/complete way to fix it from this plugin. :\

Closing.
